gnome/gtk-vnc/files/patch-meson-python3.diff

```diff
@@ -0,0 +1,11 @@
+--- src/meson.build.orig	2024-04-17 16:56:05.000000000 -0400
++++ src/meson.build	2024-04-17 16:59:36.000000000 -0400
+@@ -319,7 +319,7 @@
+   'vncdisplaykeymap.c',
+ ]
+ 
+-python = import('python').find_installation('python3')
++python = import('python').find_installation('@@PYTHON3_BIN@@')
```

Comment:
Will #23572 avoid the need for this patching? I know it works for `find_program` but I don't know about `find_installation`.

Reply:
> Will #23572 avoid the need for this patching? I know it works for `find_program` but I don't know about `find_installation`.

Possibly, wasn't aware you created that PR. Looks great though, I'll test locally with those improvements.

More to follow, with results.

Reply:
> Will #23572 avoid the need for this patching? I know it works for `find_program` but I don't know about `find_installation`.

Based on some brief testing, it doesn't appear to help: If the project is left unpatched, with the argument `python3`, it always fails regardless.

And while we technically can pass an empty string (`''`) as an argument to `find_installation`, that selects the version of Python used by Meson. Which isn't necessarily ideal either, as we then lose control of which Python is used.

Unless there's something else I'm missing, such as setting one or more Meson PG options...?

Reply:
> wasn't aware you created that PR

That's why I wanted to draw your attention to it to get some more eyeballs on it!

> it doesn't appear to help

This appeared to work for me:

```tcl
meson.native.binaries-append \
                    python=${configure.python}
```

Reply:
> This appeared to work for me:
> 
> ```tcl
> meson.native.binaries-append \
>                     python=${configure.python}
> ```

Ah gotcha. So is the expectation that we'll routinely add that to Meson-based ports? Or are you considering having that behavior enabled by default in the PG?

Reply:
My assumption is that ports would use `meson.native.binaries` to specify what binaries they want meson to find. I don't see how the portgroup could guess what each portfile wants since it will vary by portfile.

Reply:
> My assumption is that ports would use `meson.native.binaries` to specify what binaries they want meson to find. I don't see how the portgroup could guess what each portfile wants since it will vary by portfile.

Makes sense. Should we defer this until your enhancements have been merged for PG Meson?

Reply:
Either way. I have a feeling many meson ports will need to be simplified to use `meson.native.binaries` after that's merged.

Reply:
> Either way. I have a feeling many meson ports will need to be simplified to use `meson.native.binaries` after that's merged.

Sounds good, we can potentially merge as-is then. @danchr still has another 24 hours to review, and I'll defer to his preference.